pootle/apps/pootle_language/forms.py

```diff
@@ -67,32 +71,40 @@ def search(self):
 
 class LanguageTeamAdminForm(LanguageTeamBaseAdminForm):
     rm_admins = forms.ModelMultipleChoiceField(
-        label=_("Admins"),
+        # Translators: This is members with rights to manage team.
+        label=_("Administrators"),
         widget=TableSelectMultiple(item_attrs=["username"]),
         required=False,
         queryset=User.objects.none())
     rm_reviewers = forms.ModelMultipleChoiceField(
         required=False,
+        # Translators: This is members with rights to review.
         label=_("Reviewers"),
         widget=TableSelectMultiple(item_attrs=["username"]),
         queryset=User.objects.none())
     rm_submitters = forms.ModelMultipleChoiceField(
         required=False,
+        # Translators: This is members with only translate rights.
         label=_("Submitters"),
         widget=TableSelectMultiple(item_attrs=["username"]),
         queryset=User.objects.none())
     rm_members = forms.ModelMultipleChoiceField(
         required=False,
+        # Translators: This is members with only suggest rights.
         label=_("Members"),
         widget=TableSelectMultiple(item_attrs=["username"]),
         queryset=User.objects.none())
     new_member = forms.ModelChoiceField(
+        label=_("New member"),
+        help_text=_("Add some user to this team"),
         required=False,
         queryset=User.objects.none(),
         widget=forms.Select(
             attrs={
                 'class': 'js-select2-remote'}))
     role = forms.ChoiceField(
+        label=_("Role"),
+        help_text=_("Add the new member with this role"),
```

Comment:
If left blank then the field name is used as label, but won't be translatable at all. Maybe the help text is not 100% necessary, but it will help.